homeassistant/components/mqtt/notify.py

```diff
@@ -89,7 +91,7 @@ async def _async_setup_notify(
 ):
     """Set up the MQTT notify service with auto discovery."""
     config = DISCOVERY_SCHEMA(discovery_data[ATTR_DISCOVERY_PAYLOAD])
```

Comment:
```py
config: MqttNotificationConfig  = DISCOVERY_SCHEMA(discovery_data[ATTR_DISCOVERY_PAYLOAD])
```

Reply:
This does not help. We still need casting, discovery_hash was removed from the TypedDict, and cannot be accessed if we already apply the typing here.